server/matchmaker/map_pool.py

```diff
@@ -39,14 +39,21 @@ def choose_map(self, played_map_ids: Iterable[int] = ()) -> Map:
         counter.update(id_ for id_ in played_map_ids if id_ in self.maps)
 
         least_common = counter.most_common()[::-1]
-        least_count = least_common[0][1]
+        least_count = 1
+        for map_count in least_common:
+            if not (isinstance(self.maps[map_count[0]], NeroxisGeneratedMap)):
```

Comment:
What is the purpose of this? I don't think that generated maps will work with the anti-repetition logic in general because their map-ids will not exist in the database. This function is called only on data loaded from the database.

Reply:
The way it was previously least_common was all the maps with the lowest count of plays based on the list of played_map_ids. Since the generated map ids will never be in there then they would always have a cunt of 1 meaning eventually they would be the only maps in least_common

Reply:
So maybe it would be better to filter by the `Map` class instead? Since `Map` represents a map from the database, and any other types would not exist in the database.

Reply:
Aight